apps/andi/test/integration/andi_web/controllers/ingestion_controller_test.exs

```diff
@@ -132,6 +131,7 @@ defmodule Andi.IngestionControllerTest do
         |> SmartCity.Helpers.to_atom_keys()
         |> TDG.create_ingestion()
         |> struct_to_map_with_string_keys()
+        |> Map.pop("id")
```

Comment:
I think you can pass in %{id: nil} to TDG.create_ingestion to overwrite the id during the original creation instead of popping/deleting it at the end

Reply:
Doesn't that leave %{id: nil} on the ingestion? It seems like it would be closer to the actual use case of not having the field on a call to not have that left over.

Reply:
I think either would result in your new controller logic matching correctly, but yeah pop would be a true representation of how we'd like the response to come in.

nil leaves it on the struct.

```elixir
iex(1)> SmartCity.TestDataGenerator.create_ingestion(%{id: nil})
%SmartCity.Ingestion{
  allow_duplicates: true,
  ...
  id: nil,
  ...
}%
```